Library/MessageGenerator.swift

```diff
@@ -168,25 +161,47 @@ struct MessageGenerator {
         return formatter
     }()
 
-    func generateMessage(forHashValue hashValue: UInt32, sunPhase: SunPhase, yesterdayDaylightLength: Double, todayDaylightLength: Double, tomorrowDaylightLength: Double) -> Message {
-        let messageKind = Message.Kind(sunPhase: sunPhase, yesterdayDaylightLength: yesterdayDaylightLength, todayDaylightLength: todayDaylightLength, tomorrowDaylightLength: tomorrowDaylightLength)
-        switch messageKind {
-        case .longerMoreThanAMinute:
-            return longerMoreThanAMinuteMessage(for: hashValue)
-        case .longerLessThanAMinute:
-            return longerLessThanAMinuteMessage(for: hashValue)
-        case .shorterMoreThanAMinute:
-            return shorterMoreThanAMinuteMessage(for: hashValue)
-        case .shorterLessThanAMinute:
-            return self.shorterLessThanAMinuteMessage(for: hashValue)
-        case .longerTomorrowMoreThanAMinute:
-            return self.longerTomorrowMoreThanAMinuteMessage(for: hashValue)
-        case .longerTomorrowLessThanAMinute:
-            return self.longerTomorrowLessThanAMinuteMessage(for: hashValue)
-        case .shorterTomorrowMoreThanAMinute:
-            return self.shorterTomorrowMoreThanAMinuteMessage(for: hashValue)
-        case .shorterTomorrowLessThanAMinute:
-            return self.shorterTomorrowLessThanAMinuteMessage(for: hashValue)
+    func generateMessage(date: Date, hemisphere: Location.Hemisphere, sunPhase: SunPhase, daylightLenghtDifference: Double) -> Message {
+
+        if date.isSolstice {
+            if date.isJuneSolstice {
+                switch hemisphere {
+                case .southern:
+                    return Message(format: "Today is the winter solstice ☀️! From now on, every day should get a bit brighter! Enjoy!")
+                case .northern:
+                    return Message(format: "Today is the summer solstice ☀️! Days should start getting darker now…")
+                }
+            } else {
+                switch hemisphere {
+                case .southern:
+                    return Message(format: "Today is the summer solstice ☀️! Days should start getting darker now…")
+                case .northern:
+                    return Message(format: "Today is the winter solstice ☀️! From now on, every day should get a bit brighter! Enjoy!")
```

Comment:
worth pulling these two`Messages` out as variables since they're the same text in both instances?

Reply:
Even better, got a couple options from @IrisCuppen instead, so now it's the same as the others. 👍